coderd/workspaceagents.go

```diff
@@ -260,9 +260,15 @@ func (api *API) patchWorkspaceAgentStartupLogs(rw http.ResponseWriter, r *http.R
 	output := make([]string, 0)
 	level := make([]database.LogLevel, 0)
 	eof := make([]bool, 0)
-	var endLog agentsdk.StartupLog
 	outputLength := 0
-	for _, logEntry := range req.Logs {
+	for i, logEntry := range req.Logs {
+		if logEntry.EOF && i != len(req.Logs)-1 {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "EOF log must be the last log entry.",
+			})
+			return
+		}
```

Comment:
nit: not sure if it's worth it, but if you move this check before for-loop, you could fail fast

Reply:
If we do that, we need to do two iterations on the logs, first to check if any has EOF and then to construct the new format?

Reply:
Ok, I missed that. Let's leave it then.